adapter/test/stdio/stdio_adapter_test.cpp

```diff
@@ -0,0 +1,82 @@
+#include <catch_config.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <experimental/filesystem>
+#include <iostream>
+namespace fs = std::experimental::filesystem;
+
+namespace hermes::adapter::stdio::test {
+struct Arguments {
+  std::string filename = "test.dat";
+  std::string directory = "/tmp";
+  size_t request_size = 65536;
+};
+struct Info {
+  int rank = 0;
+  int comm_size = 1;
+  std::string write_data;
+  std::string read_data;
+  std::string new_file;
+  std::string existing_file;
+  size_t num_iterations = 64;
+  unsigned int offset_seed = 1;
+  unsigned int rs_seed = 1;
+  unsigned int temporal_interval_seed = 5;
+  size_t total_size;
+  size_t stride_size = 512;
+  unsigned int temporal_interval_ms = 1;
+  size_t small_min = 1, small_max = 4 * 1024;
+  size_t medium_min = 4 * 1024 + 1, medium_max = 256 * 1024;
+  size_t large_min = 256 * 1024 + 1, large_max = 3 * 1024 * 1024;
+};
+}  // namespace hermes::adapter::stdio::test
+hermes::adapter::stdio::test::Arguments args;
+hermes::adapter::stdio::test::Info info;
+
+int init() {
+  info.write_data = std::string(args.request_size, 'w');
+  info.read_data = std::string(args.request_size, 'r');
+  return 0;
+}
+int finalize() { return 0; }
+
+int pretest() {
+  fs::path fullpath = args.directory;
+  fullpath /= args.filename;
+  info.new_file = fullpath.string() + "_new";
+  info.existing_file = fullpath.string() + "_ext";
+  if (fs::exists(info.new_file)) fs::remove(info.new_file);
+  if (fs::exists(info.existing_file)) fs::remove(info.existing_file);
+  if (!fs::exists(info.existing_file)) {
+    std::string cmd = "dd if=/dev/zero of=" + info.existing_file + " bs=" +
+                      std::to_string(args.request_size * info.num_iterations) +
+                      " count=1  > /dev/null 2>&1";
+    system(cmd.c_str());
```

Comment:
On gcc 9.3 I'm getting several `unused-result` warnings everywhere you're calling `system`. Could you `REQUIRE` that the return code is valid.

Reply:
Done. This was the case across all test cases.